### ISSUE TITLE
[Note] Refactor Pull MergeBox logic

### ISSUE THREAD

**Issue by 6543** [go-gitea/gitea#11077]
currently a lot of the logic went into `templates/repo/issue/view_content/pull.tmpl` witch is general a bad idear.

move status check logick back into a go function ...
(red vs yelow vs green ... icon)  
(if if can merged or force-merged or blocked)

come up with: #11074

**Comment by 6543** [comment]
ping

**Comment by silverwind** [comment]
@6543 still valid?

**Comment by MaxWipfli** [comment]
Yes, this is still an issue. It is very related to a problem I am having, which is the following:

I want to use a default merge message (by using `.gitea/default_merge_message/MERGE_TEMPLATE.md`). However, this removes the `Reviewed-on` and `Reviewed-by` lines in the merge message, and as far as I can see, there is no way to add them back in using the message template.

The reason for this is that the logic constructing the default-default merge message (i.e., if `MERGE_TEMPLATE.md` does not exist) is embedded in [`templates/repo/issue/view_content/pull.tmpl`](https://github.com/go-gitea/gitea/blob/a80debc208fbf0ecf6ca734e454cae08adafb570/templates/repo/issue/view_content/pull.tmpl#L211C28-L211C28). If `DefaultMergeBody` (which is the templated `MERGE_TEMPLATE.md` is not set, the `Reviewed-on` and `Reviewed-by` lines are directly injected into the HTML template.

What would be desired is a way to inject these lines into the custom merge template, most likely via some variables.

**Comment by lunny** [comment]
I think the part to refactor pull mergebox logic could be closed since the related code changed a lot.
